packages/web-pkg/src/composables/resources/useCanListVersions.ts

```diff
@@ -0,0 +1,26 @@
+import { useUserStore } from '../piniaStores'
+import { isSpaceResource, isTrashResource, Resource, SpaceResource } from '@ownclouders/web-client'
+
+export const useCanListVersions = () => {
+  const userStore = useUserStore()
+
+  const canListVersions = ({ space, resource }: { space: SpaceResource; resource: Resource }) => {
+    if (resource.type === 'folder') {
+      return false
+    }
+    if (isSpaceResource(resource)) {
+      return false
+    }
+    if (isTrashResource(resource)) {
+      return false
+    }
+    if (!space.canListVersions({ user: userStore.user })) {
```

Comment:
Since this is the main business logic of the function and even matches the function name I feel like `return space.canListVersions({ user: userStore.user })` would be appropriate as last statement of the function instead of branching into a `return false` and `return true`. 😁